js/page.js

```diff
@@ -290,17 +298,40 @@
       for (let site of gGrid.sites) {
         if (site && site._link.url === message.link.url) {
           site._link.pinState = message.pinState;
+          break;
         }
       }
+      // If we are unpinning the site, update the grid.
+      if (!message.pinState) {
+        gUpdater.updateGrid(message);
+      }
     },
 
     setBlockState(message) {
       for (let site of gGrid.sites) {
         if (site && site._link.url === message.link.url) {
           site._link.blockState = message.blockState;
+          break;
```

Comment:
Nit: the above is `.filter()`ing... should fix that at some point to make the code a bit easier to read.